packages/opentelemetry-plugin-grpc-js/src/utils.ts

```diff
@@ -44,3 +50,63 @@ export const grpcStatusCodeToCanonicalCode = (
 export const grpcStatusCodeToSpanStatus = (status: number): Status => {
   return { code: status };
 };
+
+/**
+ * Returns true if the metadata contains
+ * the opentelemetry outgoing request header.
+ */
+export const containsOtelMetadata = (metadata: grpcTypes.Metadata): boolean => {
+  return metadata.get(OTEL_OUTGOING_REQUEST_KEY).length > 0;
+};
+
+/**
+ * Returns true if methodName matches pattern
+ * @param methodName the name of the method
+ * @param pattern Match pattern
+ */
+const satisfiesPattern = (
+  methodName: string,
+  pattern: IgnoreMatcher
+): boolean => {
+  if (typeof pattern === 'string') {
+    return pattern.toLowerCase() === methodName.toLowerCase();
+  } else if (pattern instanceof RegExp) {
+    return pattern.test(methodName);
+  } else if (typeof pattern === 'function') {
+    return pattern(methodName);
+  } else {
+    throw new TypeError('Pattern is in unsupported datatype');
+  }
+};
+
+/**
+ * Returns true if the current plugin configuration
+ * ignores the given method.
+ * @param methodName the name of the method
+ * @param ignoredMethods a list of matching patterns
+ * @param onException an error handler for matching exceptions
+ */
+export const methodIsIgnored = (
+  methodName: string,
+  ignoredMethods?: IgnoreMatcher[],
+  onException?: (e: Error) => void
+): boolean => {
+  if (!ignoredMethods) {
+    // No ignored rpc methods
+    return false;
+  }
+
+  try {
```

Comment:
please remove try catch, then function `satisfiesPattern` should simply return false instead of throwing an error